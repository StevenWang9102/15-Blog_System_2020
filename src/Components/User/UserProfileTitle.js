import React from "react";
import { createUseStyles } from "react-jss";
import { Link } from "react-router-dom";
const useStyles = createUseStyles({
  myButton: {
    border: "0px",
    padding: "1px 7px 1px 5px"
  },
  "& a": {
    height: "5px",
    width: "5px"
  }
});

export const UserProfileTitle = props => {
  const Button = ({ children }) => {
    const classes = useStyles();
    return (
      <button
        className={`btn btn-sm btn-outline-secondary action-btn ${classes.myButton}`}>
        {children}
      </button>
    );
  };

  const profileDetails = props.currentProfileDetail;
  return (
    <div className='user-info'>
      <div className='container'>
        <div className='row'>
          <div className='col-xs-12 col-md-10 offset-md-1'>
            <img
              src={profileDetails && profileDetails.image}
              className='user-img'
              alt='au'
            />
            <h4>{props.author_name}</h4>
            <p>{profileDetails && profileDetails.bio}</p>

            <Button>
              <span
                className='nav-link'
                onClick={() => {
                  if (props.userInformation.username) {
                    if (profileDetails.following === false)
                      props.onFollowAuthorClick(props.author_name, "POST");
                    else props.onFollowAuthorClick(props.author_name, "DELETE");
                  }
                }}>

                {props.userInformation.username === props.author_name ? (
                // Logged or not display differently
                  <span>
                    <Link to='/setting' className='nav-link'>Edit Profile Setting</Link>
                  </span>
                ) : (
                  <span>
                    {profileDetails.following
                      ? `- Unfollow ${props.author_name}  `
                      : ` + Follow ${props.author_name} Now `}
                  </span>
                )}
              </span>
            </Button>
          </div>
        </div>
      </div>
    </div>
  );
};
