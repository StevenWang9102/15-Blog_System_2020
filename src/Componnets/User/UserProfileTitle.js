import React from "react";
import { createUseStyles } from "react-jss";

const useStyles = createUseStyles({
  myButton: {
    border: "0px",
    padding: "1px 7px 1px 5px"
  },
  "& a": {
    height: "5px",
    width: "5px"
  }
});

export const UserProfileTitle = props => {
  const Button = ({ children }) => {
    const classes = useStyles();
    return (
      <button
        className={`btn btn-sm btn-outline-secondary action-btn ${classes.myButton}`}>
        {children}
      </button>
    );
  };

  return (
    <div className='user-info'>
      <div className='container'>
        <div className='row'>
          <div className='col-xs-12 col-md-10 offset-md-1'>
            <img
              src={
                props.currentProfileDetail && props.currentProfileDetail.image
              }
              className='user-img'
              alt='au'
            />
            <h4>{props.author_name}</h4>
            <p>
              {props.currentProfileDetail && props.currentProfileDetail.bio}
            </p>

            <Button>
              <i className='ion-plus-round'></i>
              <a
                className='nav-link'
                onClick={() => {
                  if (props.currentProfileDetail.following === false)
                    props.onFollowAuthorClick(props.author_name, "POST");
                  else props.onFollowAuthorClick(props.author_name, "DELETE");
                }}>
                <img src='./icon/004-settings.png' alt='' />
                {props.currentProfileDetail.following
                  ? `- Unfollow ${props.author_name}  `
                  : ` + Follow ${props.author_name} Now `}
              </a>
            </Button>
          </div>
        </div>
      </div>
    </div>
  );
};
