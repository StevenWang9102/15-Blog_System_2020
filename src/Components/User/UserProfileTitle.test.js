import React from "react";
import { UserProfileTitle } from "./UserProfileTitle";
import renderer from "react-test-renderer";

it("renders correctly", () => {
  const tree = renderer
    .create(
      <UserProfileTitle
        userInformation={{}}
        author_name='test'
        currentProfileDetail={{}}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
