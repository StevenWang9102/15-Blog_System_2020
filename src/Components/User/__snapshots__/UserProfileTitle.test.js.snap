// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="user-info"
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-xs-12 col-md-10 offset-md-1"
      >
        <img
          alt="au"
          className="user-img"
        />
        <h4>
          test
        </h4>
        <p />
        <button
          className="btn btn-sm btn-outline-secondary action-btn myButton-0-2-1"
        >
          <span
            className="nav-link"
            onClick={[Function]}
          >
            <span>
               + Follow test Now 
            </span>
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
